core/server/server.js

```diff
@@ -306,87 +297,78 @@ class Server {
 
   // See https://www.viget.com/articles/heroku-cloudflare-the-right-way/
   requireCloudflare() {
-    // Set `req.ip`, which is expected by `cloudflareMiddleware()`. This is set
-    // by Express but not Scoutcamp.
```

Comment:
So, `cloudflareMiddleware()` is supposed to handle this out of the box, though as I re-read this, I imagine that only supports `x-forwarded-for`, and that we may need to restore the `isOnFly()` part of this behavior.